src/GafferSceneUIModule/RenderPassEditorBinding.cpp

```diff
@@ -0,0 +1,707 @@
+
+class RenderPassPath : public Gaffer::Path
+{
+
+	public :
+
+		RenderPassPath( ScenePlugPtr scene, Gaffer::ContextPtr context, Gaffer::PathFilterPtr filter = nullptr )
+			:	Path( filter )
+		{
+			setScene( scene );
+			setContext( context );
+		}
+
+		RenderPassPath( ScenePlugPtr scene, Gaffer::ContextPtr context, const Names &names, const IECore::InternedString &root = "/", Gaffer::PathFilterPtr filter = nullptr )
+			:	Path( names, root, filter )
+		{
+			setScene( scene );
+			setContext( context );
+		}
+
+		IE_CORE_DECLARERUNTIMETYPEDEXTENSION( RenderPassPath, GafferSceneUI::RenderPassPathTypeId, Gaffer::Path );
+
+		~RenderPassPath() override
+		{
+		}
+
+		void setScene( ScenePlugPtr scene )
+		{
+			if( m_scene == scene )
+			{
+				return;
+			}
+
+			m_scene = scene;
+			m_plugDirtiedConnection = scene->node()->plugDirtiedSignal().connect( boost::bind( &RenderPassPath::plugDirtied, this, ::_1 ) );
+
+			emitPathChanged();
+		}
+
+		ScenePlug *getScene()
+		{
+			return m_scene.get();
+		}
+
+		const ScenePlug *getScene() const
+		{
+			return m_scene.get();
+		}
+
+		void setContext( Gaffer::ContextPtr context )
+		{
+			if( m_context == context )
+			{
+				return;
+			}
+
+			m_context = context;
+			m_contextChangedConnection = context->changedSignal().connect( boost::bind( &RenderPassPath::contextChanged, this, ::_2 ) );
+
+			emitPathChanged();
+		}
+
+		Gaffer::Context *getContext()
+		{
+			return m_context.get();
+		}
+
+		const Gaffer::Context *getContext() const
+		{
+			return m_context.get();
+		}
+
+		bool isValid( const IECore::Canceller *canceller = nullptr ) const override
+		{
+			if( !Path::isValid() )
+			{
+				return false;
+			}
+
+			const PathMatcher p = pathMatcher( canceller );
+			return p.match( names() ) & ( PathMatcher::ExactMatch | PathMatcher::DescendantMatch );
+		}
+
+		bool isLeaf( const IECore::Canceller *canceller ) const override
+		{
+			const PathMatcher p = pathMatcher( canceller );
+			const unsigned match = p.match( names() );
+			return match & PathMatcher::ExactMatch && !( match & PathMatcher::DescendantMatch );
+		}
+
+		PathPtr copy() const override
+		{
+			return new RenderPassPath( m_scene, m_context, names(), root(), const_cast<PathFilter *>( getFilter() ) );
+		}
+
+		void propertyNames( std::vector<IECore::InternedString> &names, const IECore::Canceller *canceller = nullptr ) const override
+		{
+			Path::propertyNames( names, canceller );
+			names.push_back( g_renderPassNamePropertyName );
+			names.push_back( g_renderPassEnabledPropertyName );
+		}
+
+		IECore::ConstRunTimeTypedPtr property( const IECore::InternedString &name, const IECore::Canceller *canceller = nullptr ) const override
+		{
+			if( name == g_renderPassNamePropertyName )
+			{
+				const PathMatcher p = pathMatcher( canceller );
+				if( p.match( names() ) & PathMatcher::ExactMatch )
+				{
+					return new StringData( names().back().string() );
```

Comment:
Rough summary of offline conversation: We don't anticipate people putting '/' in render pass names as they'll end up creating subdirectories in output paths so we plan to add validation to the UI to prevent these and other potentially invalid characters from being used in names. We still expect a future need to group & organise render passes into a hierarchy where the name is the leaf of the path.

To limit the bus-factor I've added a comment here: ebac8ff